### ISSUE TITLE
Feature Request: Support the "no_proxy" environment variable

### ISSUE THREAD

**Issue by georgzoeller** [request/request#853]
Currently, to bypass HTTP_PROXY for local destinations (e.g. localhost) requires specific passing special options to a request made to local.

While env.HTTP_PROXY can be used set a proxy globally or per app, it seems not possible to define an exclusion list in the same way (unless I am mistaken).

This is problematic in cases like using bower programmatically with a local registry. While you could delete process.env.HTTP_PROXY to gain access to the proxy, it would would break connectivity to the outside world at the same time. 

TL;DR: It'd be great if exclusions to env.HTTP_PROXY could be declared via the env.no_proxy environment variable. (http://www.w3.org/Daemon/User/Proxies/ProxyClients.htm, curl --noproxu)
